Documentation/Plus Codes Specification.md

```diff
@@ -0,0 +1,119 @@
+
+The padding character is defined as the zero ("0") character ([U+0030](http://unicode.org/charts/PDF/U0000.pdf)).
+
+The format separator is defined as the plus ("+") character ([U+002B](http://unicode.org/charts/PDF/U0000.pdf)).
+
+## Encoding
+
+A Plus Code uses significant digits to encode its southern and western bounds. The significant digits of a Plus Code represent latitude (north) and then longitude (east) offsets from the starting location 90° S, 180° W. 
+
+A significant digit in digit places 1, 3, 5, 7 and 9 represents an offset north equal to the north-south distance for that code length (see the next higher code length in the code length table above) multiplied by the significant digit value.
+
+A significant digit in digit places 2, 4, 6, 8 and 10 represents an offset east equal to the west-east distance for that code length multiplied by the significant digit value.
+
+A significant digit in places 11 and higher represents:
+
+* an offset north equal to the north-south distance for that code length multiplied by (the significant digit value divided by four and rounded down)
+* an offset east equal to the west-east distance for that code length multiplied by (the significant digit value modulo 4)
+
+<kbd>N</kbd><kbd>E</kbd><kbd>N</kbd><kbd>E</kbd><kbd>N</kbd><kbd>E</kbd><kbd>N</kbd><kbd>E</kbd><kbd>+</kbd><kbd>N</kbd><kbd>E</kbd><kbd>X</kbd><kbd>X</kbd><kbd>X</kbd><kbd>X</kbd><kbd>X</kbd>
+
+:information_source: Note that latitudes greater than 90° N do not exist and therefore the digit place 1 MUST only have values 0–8. Likewise, digit place 2 MUST only have values 0–17.
+
+## Full code
+
+A **full code** Plus Code is globally usable and requires no other reference to interpret.
+
+Every full code MUST include significant digits in digit places 1 up through the code length. If the code length is less than 8, then the padding character is placed in digit places after the last significant digit up to and including the 8th digit place. The format separator is added after the 8th digit place.
+
+Therefore, the set of full codes exactly matches the [Perl Compatible Regular Expression](http://pcre.org):
+
+````perl
+^[2-9C][2-9CFGHJMPQRV](0{6}\+|[2-9CFGHJMPQRVWX]{2}(0000\+|[2-9CFGHJMPQRVWX]{2}(00\+|[2-9CFGHJMPQRVWX]{2}\+([2-9CFGHJMPQRVWX]{2,7})?)))$
+````
+
+## Short code
+
+A **short code** Plus Code is meaningful only when the producer and consumer of the Plus Code can agree on the approximate latitude and longitude of a **reference location**. A short code can be easier to use and remember than a full code.
+
+Plus Codes with code length less than 8 SHALL NOT be represented as short codes.
+
+The latitudinal distance (in degrees) and longitudinal distance (in degrees) between the Plus Code area's center and the reference location determines which digit places can be omitted in the short code representation:
+
+* If both distances are less than 10 degrees, digit places 1–2 MAY be omitted.
```

Comment:
Yes! This is built into the spec under "Short codes" where it specifies the "if both distances..." part. These use half the distance as the precision amount for that code level.

I believe this amounts to 50% margin. Is that correct?

Reply:
It's true that, with an **exact** reference location agreed upon between both parties, a short code can be recovered within one full code area for any given code level (e.g. within +/- 10° if the full code was shortened by two digits). However, often the reference locations aren't exact, so to make short codes more useful many parts of the current documentation warn about pushing the limits where this isn't necessary.

For example, this wiki page mentions a safety margin of 20% (which I falsely remembered as a factor of 0.4 - should of course be 0.8): https://github.com/google/open-location-code/wiki/Guidance-for-shortening-codes

That page, by the way, is referenced from this repo's README, which also states:

> If the reference location is derived from a town or city name, it is dependent on the accuracy of the geocoding service. Although one service may place "Zurich" close to the Google office, another may move it by a hundred meters or more, and this could be enough to prevent the original code being recovered. Rather than a large city size feature to generate the reference location, it is better to use smaller, neighbourhood features, that will not have as much variation in their geocode results.
> 
> Guidelines for shortening codes are in the wiki.
> 
> Recovering shortened codes works by providing the short code and a reference location. This does not need to be the same as the location used to shorten the code, but it does need to be nearby.